javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -44,8 +44,6 @@
 
     // read http://javarevisited.blogspot.de/2014/05/double-checked-locking-on-singleton-in-java.html
     private transient volatile Supplier<? extends T> supplier;
-
-    // does not need to be volatile, visibility piggy-backs on volatile read of `supplier`
```

Comment:
Now I got what the comment meant, I though it has historical significance only.
You're right, some comment IS needed here :)